packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -0,0 +1,200 @@
+///
+/// All [DatePickerThemeData] properties are `null` by default. When null,
+/// [showDatePicker] will provide its own defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+///  * [DatePickerTheme], which describes the actual configuration of a date
+///    picker theme.
+@immutable
+class DatePickerThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [DatePickerTheme] or
+  /// [ThemeData.datePickerTheme].
+  const DatePickerThemeData({
+    this.backgroundColor,
+    this.entryModeIconColor,
+    this.helpTextStyle,
+    this.shape,
+    this.selectedDayDecoration,
+    this.disabledDayDecoration,
+    this.todayDecoration,
+  });
+
+  /// The background color of a date picker.
+  ///
+  /// If this is null, the date picker defaults to the overall theme's
+  /// [ColorScheme.background].
+  final Color? backgroundColor;
+
+  /// The color of the entry mode [IconButton].
+  ///
+  /// If this is null, the date picker defaults to:
+  /// ```
+  /// Theme.of(context).colorScheme.onSurface.withOpacity(
+  ///   Theme.of(context).colorScheme.brightness == Brightness.dark ? 1.0 : 0.6,
+  /// )
+  /// ```
+  final Color? entryModeIconColor;
+
+  /// Used to configure the [TextStyle]s for the helper text in the header.
+  ///
+  /// If this is null, the date picker defaults to the overall theme's
+  /// [TextTheme.overline].
+  final TextStyle? helpTextStyle;
+
+  /// The shape of the [Dialog] that the date picker is presented in.
+  ///
+  /// The default shape is a [RoundedRectangleBorder] with a radius of 4.0
+  final ShapeBorder? shape;
+
+  /// The box decoration of selected day.
+  final BoxDecoration? selectedDayDecoration;
+
+  /// The box decoration of disabled day.
+  final BoxDecoration? disabledDayDecoration;
+
+  /// The box decoration of today.
+  final BoxDecoration? todayDecoration;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// new values.
+  DatePickerThemeData copyWith({
+    Color? backgroundColor,
+    Color? entryModeIconColor,
+    TextStyle? helpTextStyle,
+    ShapeBorder? shape,
+    BoxDecoration? selectedDayDecoration,
+    BoxDecoration? disabledDayDecoration,
+    BoxDecoration? todayDecoration,
+  }) {
+    return DatePickerThemeData(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      entryModeIconColor: entryModeIconColor ?? this.entryModeIconColor,
+      helpTextStyle: helpTextStyle ?? this.helpTextStyle,
+      shape: shape ?? this.shape,
+      selectedDayDecoration: selectedDayDecoration ?? this.selectedDayDecoration,
+      disabledDayDecoration: disabledDayDecoration ?? this.disabledDayDecoration,
+      todayDecoration: todayDecoration ?? this.todayDecoration,
+    );
+  }
+
+  /// Linearly interpolate between two date picker themes.
+  ///
+  /// The argument `t` must not be null.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
```

Comment:
Why is this macro needed here?

Reply:
I don't know much about macros, but I found that this macro is mentioned in TimePickerTheme (see [here](https://github.com/flutter/flutter/blob/a786f65700a1038494aa17c6bd7c0a0af0881ef8/packages/flutter/lib/src/material/time_picker_theme.dart#L245))
Should I remove it ?

Reply:
Removed👍